pkg/cmd/release-payload-controller/release_creation_job_controller.go

```diff
@@ -171,9 +186,84 @@ func (c *ReleaseCreationJobController) sync(ctx context.Context, key string) err
 		return err
 	}
 
+	// Get/generate the release creation job result from the payload
+	releaseCreationJobResult := getReleaseCreationJobResult(originalReleasePayload, c.jobsNamespace)
+
+	// lookup the job
+	originalReleaseCreationJob, err := c.jobsLister.Jobs(releaseCreationJobResult.Coordinates.Namespace).Get(releaseCreationJobResult.Coordinates.Name)
+	if errors.IsNotFound(err) {
+		klog.V(4).Infof("Unable to locate release creation job: %s/%s", releaseCreationJobResult.Coordinates.Namespace, releaseCreationJobResult.Coordinates.Name)
+		err = nil
+		originalReleaseCreationJob = nil
+	}
+	if err != nil {
+		return err
+	}
+
+	if originalReleaseCreationJob != nil {
+		releaseCreationJobResult.Status = computeReleaseCreationJobStatus(originalReleaseCreationJob)
+		releaseCreationJobResult.Message = computeReleaseCreationJobMessage(originalReleaseCreationJob)
+	}
+
 	releasePayload := originalReleasePayload.DeepCopy()
 
-	klog.V(4).Infof("Syncing ReleasePayload: %s/%s", releasePayload.Namespace, releasePayload.Name)
+	klog.V(4).Infof("Syncing ReleaseCreationJobResult for ReleasePayload: %s/%s", releasePayload.Namespace, releasePayload.Name)
+	releasePayload.Status.ReleaseCreationJobResult = *releaseCreationJobResult
+	releasepayloadhelpers.CanonicalizeReleasePayloadStatus(releasePayload)
+
+	if reflect.DeepEqual(originalReleasePayload, releasePayload) {
+		return nil
+	}
+
+	_, err = c.releasePayloadClient.ReleasePayloads(releasePayload.Namespace).UpdateStatus(ctx, releasePayload, metav1.UpdateOptions{})
+	if errors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
 
 	return nil
 }
+
+func getReleaseCreationJobResult(payload *v1alpha1.ReleasePayload, jobNamespace string) *v1alpha1.ReleaseCreationJobResult {
+	result := payload.Status.ReleaseCreationJobResult.DeepCopy()
+
+	if len(result.Coordinates.Name) == 0 || len(result.Coordinates.Namespace) == 0 {
+		result = &v1alpha1.ReleaseCreationJobResult{
```

Comment:
this looks weird.  What is supposed to write this field?

Reply:
Removed